### ISSUE TITLE
"ng github-pages:deploy" is fail on 1.0.0-beta.22-1

### ISSUE THREAD

**Issue by ovrmrw** [angular/angular-cli#3507]
Error message.
```
Command failed: git clean -f -d
warning: failed to remove src/lib

There was a permissions error during git file operations, please close any open project files/folders and try again.
You might also need to return to the master branch manually.
```

---

My Environment.
```
$ ng --version
angular-cli: 1.0.0-beta.22-1
node: 6.7.0
os: win32 x64
```

---

My Solution
(from)
```
// node_modules/angular-cli/commands/github-pages-deploy.js

        function cleanGhPagesBranch() {
            return execPromise('git ls-files')
                .then(function (stdout) {
                var files = '';
                stdout.split(/\n/).forEach(function (f) {
                    // skip .gitignore & 404.html
                    if ((f != '') && (f != '.gitignore') && (f != '404.html')) {
                        files = files.concat("\"" + f + "\" ");
                    }
                });
                return execPromise("git rm -r " + files);
            });
        }
```
(to)
```
        function cleanGhPagesBranch() {
            return execPromise('git ls-files')
                .then(function (stdout) {
                var files = '';
                stdout.split(/\n/).forEach(function (f) {
                    // skip .gitignore & 404.html
                    if ((f != '') && (f != '.gitignore') && (f != '404.html')) {
                        files = files.concat("\"" + f + "\" ");
                    }
                });
                // return execPromise("git rm -r " + files);
                if (files) {
                    return execPromise("git rm -r " + files);
                } else {
                    return execPromise('git ls-files'); // do something
                }
            });
        }
```

It works.

**Comment by arm4design** [comment]
Fix above worked at my end

Thanks @ovrmrw !!

**Comment by taylorjb** [comment]
I'm responsible for this and am truly sorry.  I wish it hadn't been included in `1.0.0-beta.22-1`.

It has been fixed as of `1.0.0-beta.23` with the [following](https://github.com/angular/angular-cli/blob/v1.0.0-beta.23/packages/angular-cli/commands/github-pages-deploy.ts#L220-L223):

```typescript
return execPromise(`git rm -r ${files}`)
  .catch(() => {
    // Ignoring errors when trying to erase files.
  });
```

The issue is simply that `git rm -r` errors if its not provided any file names. This only happens upon initial deployment as there are no old files yet to remove.

**Comment by filipesilva** [comment]
Closed via https://github.com/angular/angular-cli/pull/4385.
